tables_script.sql

```diff
@@ -63,14 +64,21 @@ CREATE TABLE IF NOT EXISTS `biosero_uat`.`destination_plate_wells` (
   `id` INT NOT NULL AUTO_INCREMENT,
   `automation_system_run_id` INT NOT NULL,
   `barcode` VARCHAR(255) NOT NULL,
-  `coordinate` VARCHAR(255) NOT NULL,
+  `coordinate` ENUM('A1','A2','A3','A4','A5','A6','A7','A8','A9','A10','A11','A12','B1','B2','B3','B4','B5','B6','B7','B8','B9','B10','B11','B12','C1','C2','C3','C4','C5','C6','C7','C8','C9','C10','C11','C12','D1','D2','D3','D4','D5','D6','D7','D8','D9','D10','D11','D12','E1','E2','E3','E4','E5','E6','E7','E8','E9','E10','E11','E12','F1','F2','F3','F4','F5','F6','F7','F8','F9','F10','F11','F12','G1','G2','G3','G4','G5','G6','G7','G8','G9','G10','G11','G12','H1','H2','H3','H4','H5','H6','H7','H8','H9','H10','H11','H12') NOT NULL,
   `source_plate_well_id` INT DEFAULT NULL,
   `control_plate_well_id` INT DEFAULT NULL,
   `created_at` DATETIME DEFAULT CURRENT_TIMESTAMP NOT NULL,
   `updated_at` DATETIME DEFAULT CURRENT_TIMESTAMP NOT NULL,
   PRIMARY KEY (`id`),
   KEY `index_destination_plate_wells_on_barcode` (`barcode`),
   CONSTRAINT `run_destination_plate_well` UNIQUE (`automation_system_run_id`,`barcode`,`coordinate`),
+  CONSTRAINT `source_plate_well_id_or_control_plate_well_id` CHECK (
```

Comment:
@harrietc52, I haven't tried it myself but would [this](https://dev.mysql.com/doc/refman/8.0/en/logical-operators.html#operator_xor) work?

Reply:
Yep XOR, I'll have a look into it @pjvv

Reply:
Could replace with
```sql
  CONSTRAINT `source_plate_well_id_or_control_plate_well_id` CHECK (
	`control_plate_well_id` = 1 XOR `source_plate_well_id` = 1   
  )
```

Reply:
changed